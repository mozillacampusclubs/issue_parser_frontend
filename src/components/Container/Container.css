.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-intro {
  font-size: large;
}

.side-search-bar {
  height: 100%;
  padding-left: 30px;
  border-right: 7px solid black;
}
