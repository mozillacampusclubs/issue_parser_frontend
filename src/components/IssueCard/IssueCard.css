.issue-card-date {
  margin-top: 8px;
  font-size: 0.9em;
}
