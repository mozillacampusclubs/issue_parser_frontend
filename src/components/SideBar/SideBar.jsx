import React, { Component } from 'react';
import { connect } from 'react-redux';
import Select from 'react-select';
import 'react-select/dist/react-select.css';
import { fetchMetaData, fetchIssuesList } from '../../redux/actions/issuesActions';

import './SideBar.css';

// Use named export for unconnected component (for tests)
export class SideBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      language: '',
      tech_stack: '',
      experience_needed: '',
      regions: this.props.regionId,
    };
    this.props.dispatch(fetchMetaData());
    this.handleChange = this.handleChange.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.props.dispatch(fetchIssuesList(this.state));
  }

  handleChange(name, value) {
    let change = {};
    change[name] = value.value;
    this.setState(change);
  }

  makeOptionsList(list) {
    let optionsList = {};
    optionsList = list.map((option, key) => {
      return {value: option, label: option}
    });
    return optionsList;
  }

  render() {
    const metadata = this.props.metadata;
    const languages = metadata.data.language;
    const experienceNeeded = metadata.data.experience_needed;
    const techStack = metadata.data.tech_stack;

    const language_label = 'language';
    const experience_needed_label = 'experience_needed';
    const tech_stack_label = 'tech_stack';

    let renderingComponent;

    if (metadata.fetched) {
      renderingComponent = (
        <div>
          <form className="form-group" >
            <div className="form-group">
              <h5>Language</h5>
              <Select
                autofocus
                name="form-field-name"
                clearable={false}
                value={this.state.language}
                options={this.makeOptionsList(languages)}
                onChange={this.handleChange.bind(this, language_label)}
              />
            </div>
            <div className="form-group">
              <h5>Experience Level</h5>
              <Select
                autofocus
                name="form-field-name"
                clearable={false}
                value={this.state.experience_needed}
                options={this.makeOptionsList(experienceNeeded)}
                onChange={this.handleChange.bind(this, experience_needed_label)}
              />
            </div>
            <div className="form-group">
              <h5>Technology Stack</h5>
              <Select
                autofocus
                name="form-field-name"
                value={this.state.tech_stack}
                clearable={false}
                options={this.makeOptionsList(techStack)}
                onChange={this.handleChange.bind(this, tech_stack_label)}
              />
            </div>
          </form>
          <button className="btn btn-primary" onClick={ e => this.handleClick() }>Submit</button>
        </div>
      );
    } else if (metadata.fetching) {
      renderingComponent = <h3>Loading... </h3>
    } else {
      renderingComponent = <h4 className="alert alert-warning">Error: Something went wrong</h4>;
    }

    return (
      <div>
        {renderingComponent}
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    metadata: state.issues.metadata,
  };
}

// Use default export for the connected component (for app)
export default connect(mapStateToProps)(SideBar);
