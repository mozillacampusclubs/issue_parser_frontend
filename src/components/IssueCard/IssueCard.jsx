import React, { Component } from 'react';
import dateFormat from 'dateformat';

import './IssueCard.css';

// Use named export for unconnected component (for tests)
export class IssueCard extends Component {
  formatDate(date) {
    return dateFormat(date, "dddd, mmmm dS, yyyy");
  }

  render() {
    const data = this.props.data;
    let labelsList;

    labelsList = data.issue_labels.map((labelInstance, key) => {
      return (
        <span className="label label-default issue-card-labels" key={key} style={{backgroundColor: '#' + labelInstance.label_color}}>
          {labelInstance.label_name}
        </span>
      );
    });

    return (
      <div>

        <div className="panel panel-default">
          <div className="panel-heading">
            <h4 className="panel-title">
              <b><a href={data.issue_url} target="_">{data.title}</a></b>
            </h4>
            <div className="lightslategray-text issue-card-date">
              <span>
                Created: <b>{this.formatDate(data.created_at)}</b>
              </span>
              <span className="pull-right">
                Last Updated: <b>{this.formatDate(data.updated_at)}</b>
              </span>
            </div>
          </div>

          <div className="panel-body">
            <div className="gray-text">
              {labelsList}
            </div>
            <br/>
            <div className="row">
              <div className="col-md-6">
                <div className="gray-text">
                  <span>Experience Required: </span>
                  <span className="issue-card-attr"><b>{data.experience_needed}</b></span>
                </div>
                <br/>
                <div className="gray-text">
                  <span>Language: </span>
                  <span className="issue-card-attr"><b>{data.language}</b></span>
                </div>
              </div>
              <div className="col-md-6">
                <div className="gray-text">
                  <span>Technology Stack: </span>
                  <span className="issue-card-attr"><b>{data.tech_stack}</b></span>
                </div>
                <br/>
                <div className="gray-text">
                  <span>Expected Time: </span>
                  <span className="issue-card-attr"><b>{data.expected_time}</b></span>
                </div>
              </div>
            </div>
          </div>
        </div>

      </div>
    );
  }
}

// Use default export for the connected component (for app)
export default IssueCard;
