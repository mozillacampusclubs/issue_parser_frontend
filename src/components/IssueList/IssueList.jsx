import React, { Component } from 'react';
import { DropdownButton, MenuItem } from 'react-bootstrap';

import { connect } from 'react-redux';
import { sortIssues } from '../../redux/actions/issuesActions';
import IssueCard from '../IssueCard/IssueCard'

// Use named export for unconnected component (for tests)
export class IssueList extends Component {
  constructor() {
    super();
    this.state = {
      ordering : 'random',
    }
    this.sortIssues = this.sortIssues.bind(this);
  }

  sortIssues(value, order) {
    this.props.dispatch(sortIssues(value, order));
    let sort = order === 1 ? ' (ascending)' : ' (descending)';
    this.setState({ordering: value + sort});
  }

  render() {
    const issuesList = this.props.issuesList;
    let renderingComponent = <h1>Be With Us! :)</h1>;
    let IssueCardList;

    if (issuesList.fetched) {
      IssueCardList = issuesList.data.map((issue, key) => {
        return <IssueCard key={key} data={issue} />
      });
      renderingComponent = (
        <div>
          <div>
            <h3>
              <b className="gray-text">Issues</b>
              <div className="pull-right">
                <span className="badge">Ordering: {this.state.ordering} </span>
                <DropdownButton bsSize="small" bsStyle="info" title="Sort" id="sort">
                  <MenuItem id="sort-asc-experience_needed" onClick={e => this.sortIssues("experience_needed", 1)} eventKey="1">Exp-level (ascending)</MenuItem>
                  <MenuItem id="sort-desc-experience_needed" onClick={e => this.sortIssues("experience_needed", -1)} eventKey="2">Exp-level (descending)</MenuItem>
                  <MenuItem divider />
                  <MenuItem id="sort-asc-expected_time" onClick={e => this.sortIssues("expected_time", 1)} eventKey="3">Expected time (descending)</MenuItem>
                  <MenuItem id="sort-desc-expected_time" onClick={e => this.sortIssues("expected_time", -1)} eventKey="4">Expected time (descending)</MenuItem>
                </DropdownButton>
              </div>
            </h3>
          </div>
          <div>
            {IssueCardList}
          </div>
        </div>
      );
    } else if (issuesList.fetching) {
      renderingComponent = (
        <div>
          <br/><br/>
          <h4>Loading...</h4>
        </div>
      );
    } else if (issuesList.error) {
      renderingComponent = (
        <div>
          <br/>
          <div className="alert alert-warning ">Something went wrong</div>
        </div>
      );
    }

    return (
      <div>
        {renderingComponent}
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    issuesList: state.issues.issuesList,
  };
}

// Use default export for the connected component (for app)
export default connect(mapStateToProps)(IssueList);
