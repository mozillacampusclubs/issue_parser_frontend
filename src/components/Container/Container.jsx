import React, { Component } from 'react';
import 'bootstrap/dist/css/bootstrap.css';
import './Container.css';
import SideBar from '../SideBar/SideBar.jsx';
import IssueList from '../IssueList/IssueList.jsx';

class Container extends Component {
  render() {
    return (
      <div className="container">
        <section className="row">
          <section className="col-md-3 side-bar">
            <br/>
            <SideBar />
          </section>
          <section className="col-md-9">
            <IssueList />
          </section>
        </section>
        <footer></footer>
      </div>
    );
  }
}

export default Container;
