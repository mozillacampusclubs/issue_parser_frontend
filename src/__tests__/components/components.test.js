import React from 'react';
import { shallow, mount } from 'enzyme';
import { Container } from '../../components/Container/Container';
import { IssueCard } from '../../components/IssueCard/IssueCard';
import { SideBar } from '../../components/SideBar/SideBar';
import { IssueList } from '../../components/IssueList/IssueList';
import ReactShallowRenderer from 'react-test-renderer/shallow'; // For snapshot testing

describe('container component', () => {
  it('renders correctly', () => {
    const renderer = new ReactShallowRenderer();
    expect(renderer.render(<Container />)).toMatchSnapshot();
  });
})


describe('issue card component', () => {
  const mockData = {
    "issue_id": 233564738,
    "title": "Dockerize Project",
    "experience_needed": "easyfix",
    "expected_time": "3 hours",
    "language": "python",
    "tech_stack": "django",
    "created_at": "2017-06-05T11:47:01Z",
    "updated_at": "2017-06-06T10:35:57Z",
    "issue_number": 7,
    "issue_labels": [
      {
        "label_id": 613678729,
        "label_name": "enhancement",
        "label_color": "84b6eb",
        "label_url": "https://api.github.com/repos/mozillacampusclubs/issue_parser_backend/labels/enhancement"
      }
    ],
    "issue_url": "https://github.com/mozillacampusclubs/issue_parser_backend/issues/7",
    "issue_body": "",
  }

  it('renders correctly', () => {
    const renderer = new ReactShallowRenderer();
    expect(renderer.render(<IssueCard data={mockData} />)).toMatchSnapshot();
  })
})

describe('side bar component', () => {
  const props = {
    dispatch: jest.fn(),
    metadata: {
      data: {
        "experience_needed": [
          "senior",
          "moderate",
          "easyfix"
        ],
        "language": [
          "python",
          "javascript",
        ],
        "tech_stack": [
          "react.js",
          "django, celery",
          "jest",
          "django"
        ]
      },
      fetched: false,
      fetching: false,
      error: false
    }
  }
  it('renders correctly when data is fetching', () => {
    const metadata = {...props.metadata, fetching: true};
    const newProps = {...props, metadata}
    const renderer = new ReactShallowRenderer();
    expect(renderer.render(<SideBar {...newProps} />)).toMatchSnapshot();
  });

  it('renders correctly when data is fetched', () => {
    const metadata = {...props.metadata, fetched: true};
    const newProps = {...props, metadata}
    const renderer = new ReactShallowRenderer();
    expect(renderer.render(<SideBar {...newProps} />)).toMatchSnapshot();
  });

  it('renders correctly when server error', () => {
    const metadata = {...props.metadata, error: true};
    const newProps = {...props, metadata}
    const renderer = new ReactShallowRenderer();
    expect(renderer.render(<SideBar {...newProps} />)).toMatchSnapshot();
  });
})

describe('issue list component', () => {
  const props = {
    dispatch: jest.fn(),
    issuesList: {
      data: [],
      fetched: false,
      fetching: false,
      error: false
    }
  }

  it('renders correctly when data is fetching', () => {
    const issuesList = {...props.issuesList, fetching: true};
    const newProps = {...props, issuesList}
    const renderer = new ReactShallowRenderer();
    expect(renderer.render(<IssueList {...newProps} />)).toMatchSnapshot();
  });

  it('renders correctly when data is fetched', () => {
    const issuesList = {...props.issuesList, fetched: true};
    const newProps = {...props, issuesList}
    const renderer = new ReactShallowRenderer();
    expect(renderer.render(<IssueList {...newProps} />)).toMatchSnapshot();
  });

  it('renders correctly when server error', () => {
    const issuesList = {...props.issuesList, error: true};
    const newProps = {...props, issuesList}
    const renderer = new ReactShallowRenderer();
    expect(renderer.render(<IssueList {...newProps} />)).toMatchSnapshot();
  });
})


