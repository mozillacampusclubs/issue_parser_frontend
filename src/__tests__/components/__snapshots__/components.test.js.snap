exports[`container component renders correctly 1`] = `
<div
  className="container">
  <section
    className="row">
    <section
      className="col-md-3 side-bar">
      <br />
      <Connect(SideBar) />
    </section>
    <section
      className="col-md-9">
      <Connect(IssueList) />
    </section>
  </section>
  <footer />
</div>
`;

exports[`issue card component renders correctly 1`] = `
<div>
  <div
    className="panel panel-default">
    <div
      className="panel-heading">
      <h4
        className="panel-title">
        <b>
          <a
            href="https://github.com/mozillacampusclubs/issue_parser_backend/issues/7"
            target="_">
            Dockerize Project
          </a>
        </b>
      </h4>
      <div
        className="lightslategray-text issue-card-date">
        <span>
          Created: 
          <b>
            Monday, June 5th, 2017
          </b>
        </span>
        <span
          className="pull-right">
          Last Updated: 
          <b>
            Tuesday, June 6th, 2017
          </b>
        </span>
      </div>
    </div>
    <div
      className="panel-body">
      <div
        className="gray-text">
        <span
          className="label label-default"
          style={
            Object {
              "backgroundColor": "#84b6eb",
            }
          }>
          enhancement
        </span>
      </div>
      <br />
      <div
        className="row">
        <div
          className="col-md-6">
          <div
            className="gray-text">
            <span>
              Experience Required: 
            </span>
            <span
              className="issue-card-attr">
              <b>
                easyfix
              </b>
            </span>
          </div>
          <br />
          <div
            className="gray-text">
            <span>
              Language: 
            </span>
            <span
              className="issue-card-attr">
              <b>
                python
              </b>
            </span>
          </div>
        </div>
        <div
          className="col-md-6">
          <div
            className="gray-text">
            <span>
              Technology Stack: 
            </span>
            <span
              className="issue-card-attr">
              <b>
                django
              </b>
            </span>
          </div>
          <br />
          <div
            className="gray-text">
            <span>
              Expected Time: 
            </span>
            <span
              className="issue-card-attr">
              <b>
                3 hours
              </b>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`issue list component renders correctly when data is fetched 1`] = `
<div>
  <div>
    <div>
      <h3>
        <b
          className="gray-text">
          Issues
        </b>
        <div
          className="pull-right">
          <span
            className="badge">
            Ordering: 
            random
             
          </span>
          <DropdownButton
            bsSize="small"
            bsStyle="info"
            id="sort"
            title="Sort">
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey="1"
              header={false}
              onClick={[Function]}>
              Exp-level (ascending)
            </MenuItem>
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey="2"
              header={false}
              onClick={[Function]}>
              Exp-level (descending)
            </MenuItem>
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={true}
              header={false} />
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey="3"
              header={false}
              onClick={[Function]}>
              Expected time (descending)
            </MenuItem>
            <MenuItem
              bsClass="dropdown"
              disabled={false}
              divider={false}
              eventKey="4"
              header={false}
              onClick={[Function]}>
              Expected time (descending)
            </MenuItem>
          </DropdownButton>
        </div>
      </h3>
    </div>
    <div>
      
    </div>
  </div>
</div>
`;

exports[`issue list component renders correctly when data is fetching 1`] = `
<div>
  <div>
    <br />
    <br />
    <h4>
      Loading...
    </h4>
  </div>
</div>
`;

exports[`issue list component renders correctly when server error 1`] = `
<div>
  <div>
    <br />
    <div
      className="alert alert-warning ">
      Something went wrong
    </div>
  </div>
</div>
`;

exports[`side bar component renders correctly when data is fetched 1`] = `
<div>
  <div>
    <form
      className="form-group">
      <div
        className="form-group">
        <h5>
          Language
        </h5>
        <input
          className="form-control"
          list="languages"
          onChange={[Function]}
          value={undefined} />
        <datalist
          id="languages">
          <option
            value="python" />
          <option
            value="javascript" />
        </datalist>
      </div>
      <div
        className="form-group">
        <h5>
          Experience Level
        </h5>
        <input
          className="form-control"
          list="experienceNeeded"
          onChange={[Function]}
          value={undefined} />
        <datalist
          id="experienceNeeded">
          <option
            value="senior" />
          <option
            value="moderate" />
          <option
            value="easyfix" />
        </datalist>
      </div>
      <div
        className="form-group">
        <h5>
          Technology Stack
        </h5>
        <input
          className="form-control"
          list="techStack"
          onChange={[Function]}
          value={undefined} />
        <datalist
          id="techStack">
          <option
            value="react.js" />
          <option
            value="django, celery" />
          <option
            value="jest" />
          <option
            value="django" />
        </datalist>
      </div>
    </form>
    <button
      className="btn btn-primary"
      onClick={[Function]}>
      Submit
    </button>
  </div>
</div>
`;

exports[`side bar component renders correctly when data is fetching 1`] = `
<div>
  <h3>
    Loading... 
  </h3>
</div>
`;

exports[`side bar component renders correctly when server error 1`] = `
<div>
  <h4
    className="alert alert-warning">
    Error: Something went wrong
  </h4>
</div>
`;
